pkg/clientwrappers/usergroup/usergroup.go

```diff
@@ -192,6 +205,22 @@ func updateUserGroupEntity(userGroup *v1.NifiUserGroup, users []*v1.NifiUser, en
 	for _, user := range users {
 		entity.Component.Users = append(entity.Component.Users, nigoapi.TenantEntity{Id: user.Status.Id})
 	}
+
+	n := 0
+	for _, tenant := range entity.Component.Users {
+		Found := false
+		for _, user := range users {
+			if user.Status.Id == tenant.Id {
+				Found = true
+				break
+			}
+		}
+		if Found {
+			entity.Component.Users[n] = tenant
+			n++
+		}
+	}
+	entity.Component.Users = entity.Component.Users[:n]
```

Comment:
As we loop through the users before to add all of them in the group existing list and you remove the excess with your update, wouldn't it be better to reset the list to an empty list before adding the user. Like just add `entity.Component.Users = make([]nigoapi.TenantEntity, 0)` before the `for _, user := range users` loop?